docker/scripts/New-AzureStackTestResources.ps1

```diff
@@ -0,0 +1,127 @@
+        -MaxVirtualNetworkGatewayConnectionsPerSubscription $connectionsPerSubscription `
+        -MaxVnetsPerSubscription $vNetsPerSubscription `
+        -MaxVirtualNetworkGatewaysPerSubscription $gatewaysPerSubscription `
+        -MaxSecurityGroupsPerSubscription $securityGroupsPerSubscription `
+        -MaxLoadBalancersPerSubscription $loadBalancersPerSubscription
+    $serviceQuotas += $networkQuota.Id
+
+    $keyvaultQuota = Get-AzsKeyVaultQuota -Location $Location
+    $serviceQuotas += $keyvaultQuota.Id
+
+    $serviceQuotas
+}
+
+function Set-Offer
+{
+    $getResourceGroup = Get-AzResourceGroup -Name $resourceGroup
+    if ($null -eq $getResourceGroup)
+    {
+        New-AzResourceGroup -Name $resourceGroup -Location $Location
+    }
+
+    $getPlan = Get-AzsPlan -ResourceGroupName $resourceGroup -Name $plan
+    if ($null -eq $getPlan)
+    {
+        $serviceQuotas = New-Quotas
+        $getPlan = New-AzsPlan -Name $plan `
+            -DisplayName $plan `
+            -Location $Location `
+            -ResourceGroupName $resourceGroup `
+            -QuotaIds $serviceQuotas `
+            -ErrorAction Stop
+    }
+    $getPlan | Out-String
+    
+    $getOffer = Get-AzsManagedOffer -ResourceGroupName $resourceGroup -Name $offer
+    if ($null -eq $getOffer)
+    {
+        $getOffer = New-AzsOffer -Name $offer `
+            -DisplayName $offer `
+            -BasePlanIds @($getPlan.Id) `
+            -Location $Location `
+            -ResourceGroupName $resourceGroup `
+            -ErrorAction Stop
+    }
+    $getOffer = Get-AzsManagedOffer -ResourceGroupName $resourceGroup -Name $offer 
+    $getOffer.state = "Public"
+    $getOffer | Set-AzsOffer | Out-String
+}
+
+function Register-Services
+{
+    Write-Verbose "Registering resource providers for tenant admin."
+    ("Microsoft.Storage", "Microsoft.Compute", "Microsoft.Network", "Microsoft.KeyVault") | ForEach-Object {Get-AzResourceProvider -ProviderNamespace $_} | Register-AzResourceProvider | Out-Null
+    $sleepTime = 0
+    while ($true) {
+        $sleepTime += 10
+        Start-Sleep -Seconds  10
+        $requiredRPs = Get-AzResourceProvider -ListAvailable | Where-Object {$_.ProviderNamespace -in ("Microsoft.Storage", "Microsoft.Compute", "Microsoft.Network", "Microsoft.KeyVault")}
+        $notRegistered = $requiredRPs | Where-Object {$_.RegistrationState -ne "Registered"}
+        $registered = $requiredRPs | Where-Object {$_.RegistrationState -eq "Registered"}
+        if (($sleepTime -ge 120) -and $notRegistered) {
+            Get-AzResourceProvider | Format-Table
+            LogInfo("Exception: Resource providers did not get registered in time")
+            throw [System.Exception] "Resource providers did not get registered in time."
+        }
+        elseif ($registered.Count -eq $requiredRPs.Count) {
+            break
+        }
+    }
+    Get-AzResourceProvider | Format-Table -AutoSize
+}
+
+function Main
+{
+    Connect-ServicePrincipal
```

Comment:
Forgot to remove this line after removing the method since it is expected the user is logged in when this script is run.